internal/controllers/machineset/machineset_controller.go

```diff
@@ -362,8 +362,21 @@ func (r *Reconciler) syncMachines(ctx context.Context, machineSet *clusterv1.Mac
 	log := ctrl.LoggerFrom(ctx)
 	for i := range machines {
 		m := machines[i]
-		// If the machine is already being deleted, we don't need to update it.
+		// If the machine is already being deleted, we only need to sync
+		// the subset of fields that impact tearing down a machine
 		if !m.DeletionTimestamp.IsZero() {
+			patch := client.MergeFrom(m.DeepCopy())
+
+			// Set all other in-place mutable fields that impact the ability to tear down existing machines.
+			m.Spec.NodeDrainTimeout = machineSet.Spec.Template.Spec.NodeDrainTimeout
+			m.Spec.NodeDeletionTimeout = machineSet.Spec.Template.Spec.NodeDeletionTimeout
+			m.Spec.NodeVolumeDetachTimeout = machineSet.Spec.Template.Spec.NodeVolumeDetachTimeout
+
+			err := r.Client.Patch(ctx, m, patch)
+			if err != nil {
+				log.Error(err, "failed to update Machine", "Machine", klog.KObj(m))
```

Comment:
nit: can we use upper case here for log.Error(... according to https://github.com/kubernetes-sigs/cluster-api/pull/10613 cc @sbueringer

Reply:
A better reference: https://cluster-api.sigs.k8s.io/developer/logging.html#log-messages :)

Reply:
yep, no problem, done!